src/app/components/calendar/calendarForm.tsx

```diff
@@ -25,34 +26,27 @@ const CalendarForm = () => {
     color: "blackAndWhite",
   };
 
-  const [calendarSetup, setCalendar] = useState<CalendarValuesProps | null>(null);
-
-  const [calendarDates, setCalendarDates] = useState({
+  const calendarDatesDefault = {
     weekNumber: null,
     weekDayNumbers: [],
     month: new Date().getMonth(),
     year: new Date().getFullYear(),
-  });
+  }
+
+  const [calendarSetup, setCalendar] = useLocalStorage('calendarData', calendarDefaultValues)
+  const [calendarDates, setCalendarDates] = useLocalStorage('calendarDatesData', calendarDatesDefault)
+
+  const [mount, setMount]= useState(false)
 
   // TMP
-  const handleInput = (e: any) => {
+  const handleInput = (e: { target: { [key: string]: any }}) => {
     const fieldName = e.target.name;
     const fieldValue = e.target.value;
 
-    setCalendar((prevState: any) => ({
-      ...prevState,
-      [fieldName]: fieldValue,
-    }));
-
-    let newVals = (prevState: any) => {
-      return { ...prevState, [fieldName]: fieldValue };
-    };
-    let values = newVals(calendarSetup);
-
-    localStorage.setItem("calendarData", JSON.stringify(values));
+    setCalendar({ ...calendarSetup, [fieldName]: fieldValue })
   };
 
-  const setCalendarHandle = (e: any) => {
+  const setCalendarHandle = (e: { target: { [key: string]: any }}) => {
```

Comment:
Also unrelated to this MR, but I have several observations about this function.

0. What is a calendar handle? Would `setCalendarState` work better?

1. The function is typed as if it expected a user event (ie. an instance of the [Event interface](https://developer.mozilla.org/en-US/docs/Web/API/Event), having a target prop etc.), However, it does not. I would recommend giving it params propName and value instead, since the "target" prop is useless here. Wdyt?

```suggestion
  const setCalendarState = (propName, value) => {
```

Even better, let's make accept the entire state of the calendar:

```suggestion
  const setCalendarState = (state: CalendarState) => {
```

There is also an issue of representation of the calendar state. All the properties (weekNumber, weekDayNumbers, month and year) are computable from the currently selected date. I would consider representing the state of the calendar as a single Date object, and computing the year, week and so on where you need it.

For example, you could have a single helper function for computing the days of the week that contains a date.
I see several places where you're using `Array.from({length: 7}, (_, i) => ...`, ideally this would be unified:

```
const getDaysOfWeek = (aDayInTheWeek: Date) => {
  // Let's use `Array.from({length: 7}, (_, i) => ...` here and nowhere else, if possible.
};
```

This function would be useful even if you decide to keep the current representation of calendar state.

Does this make sense to you?